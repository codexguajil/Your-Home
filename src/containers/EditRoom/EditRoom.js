import React from 'react';
import { Button, Modal, Form, Input } from 'antd';

const CollectionCreateForm = Form.create({ name: 'form_in_modal' })(
  class extends React.Component {
    render() {
      const { visible, onCancel, onCreate, form } = this.props;
      const { getFieldDecorator } = form;
      return (
        <Modal
          visible={visible}
          title="Edit Room"
          okText="Update"
          onCancel={onCancel}
          onOk={onCreate}
        >
          <Form layout="vertical">
            <Form.Item label="Type">
              {getFieldDecorator('type', {
                rules: [{ required: false, message: 'Please input the type of room!' }],
              })(<Input placeholder='Update here, or leave blank to keep unchanged' />)}
            </Form.Item>
            <Form.Item label="Name">
              {getFieldDecorator('name', {
                rules: [{ required: false, message: 'Please input the name of the room!' }],
              })(<Input placeholder='Update here, or leave blank to keep unchanged' />)}
            </Form.Item>
            <Form.Item label="Description">
              {getFieldDecorator('description')(<Input type="textarea" placeholder='Update here, or leave blank to keep unchanged' />)}
            </Form.Item>
            <Form.Item className="collection-create-form_last-form-item">
              {getFieldDecorator('modifier', {
                initialValue: 'public',
              })}
            </Form.Item>
          </Form>
        </Modal>
      );
    }
  },
);

class EditRoom extends React.Component {
  state = {
    visible: false,
  };

  showModal = () => {
    this.setState({ visible: true });
  };

  handleCancel = () => {
    this.setState({ visible: false });
  };

  handleCreate = () => {
    const form = this.formRef.props.form;
    form.validateFields((err, values) => {
      if (err) {
        return;
      }

      console.log('Received values of form: ', values);
      form.resetFields();
      this.setState({ visible: false });
    });
  };

  saveFormRef = formRef => {
    this.formRef = formRef;
  };

  render() {
    return (
      <div className="edit-room-btn">
        <Button onClick={this.showModal} size="small" type="link">
        <i className="fas fa-pen"></i>
        </Button>
        <CollectionCreateForm
          wrappedComponentRef={this.saveFormRef}
          visible={this.state.visible}
          onCancel={this.handleCancel}
          onCreate={this.handleCreate}
        />
      </div>
    );
  }
}

export default EditRoom;