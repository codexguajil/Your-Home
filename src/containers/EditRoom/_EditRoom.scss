.edit-room-btn {
 .fa-pen {
   color: #1790ff;
 }
}