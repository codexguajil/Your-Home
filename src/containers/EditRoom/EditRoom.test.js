import React from 'react';
import { shallow } from 'enzyme';
import EditRoom from './EditRoom';

describe('EditRoom', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(<EditRoom/>)
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('showModal', () => {
    wrapper.instance().showModal();
    expect(wrapper.state('visible')).toBe(true);
  });

  it('handleCreate', () => {
    wrapper.instance().handleCancel();
    expect(wrapper.state('visible')).toBe(false);
  });

  it('saveFormRef', () => {

  });

  it('handleCreate', () => {

  });

});