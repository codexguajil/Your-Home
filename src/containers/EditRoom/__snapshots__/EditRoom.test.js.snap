// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditRoom should match snapshot 1`] = `
<div
  className="edit-room-btn"
>
  <Button
    block={false}
    ghost={false}
    htmlType="button"
    loading={false}
    onClick={[Function]}
    size="small"
    type="link"
  >
    <i
      className="fas fa-pen"
    />
  </Button>
  <Form(Component)
    editRoom={[Function]}
    onCancel={[Function]}
    onCreate={[Function]}
    visible={false}
  />
</div>
`;
